### PR TITLE
Hannah - API Muncher

### PR DESCRIPTION
**Pull request by hertweckhr1**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, and how did you try querying the API? | I explored by first checking out the Edamam API specs and what was needed.  I then went into Postman to check out the api and query through it to learn more. 
What is an API Wrapper? Why is it in `lib`? How would your project change if you needed to interact with more than one API (aka more than just the Edamam API)? | An API Wrapper is a place to put information into your app to access an API.  An API Wrapper is meant to make life easier so you have one place to reference the API in your app without having to constantly repeat yourself. 
Describe your API wrapper, the methods you created in it, and why you decided to create those methods/how they were helpful | My API Wrapper allows the app to list recipes based on a search result and it allows to show a specific recipe based on a unique id. 
What was an edge case or failure case test you wrote for your API Wrapper? What was a nominal case? | One nominal case I wrote in my API Wrapper was that is could show a recipe for an existing recipe.  An edge case would be, it cannot show infor a recipe with an invalid id"
How does VCR aid in testing an API? | VCR allow you to record your action with an API, letting you replay it later - this reduces cost and improves stability for testing. 
What is the Heroku URL of your deployed application? | https://muncher-mania.herokuapp.com/

### CODE REVIEW

File: app/views/layouts/not_found.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div class="not-found text-center container">
```
Comment (CheezItMan) [inline app/views/layouts/not_found.html.erb:1]: I think `not_found` would work much better as a view instead of a layout.

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  </div>
+</div>
+
+</header>
+
+<% if @recipes %>
+<div class="container index-container mx-auto">
+  <% @recipes.each do |recipe| %>
+    <div class="card index-card bg-white">
+      <%= image_tag(recipe.image, alt: "recipe image", class: "card-img-top" ) %>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:18]: I suggest making this image a link as well.  It's human nature to want to click on the images.

File: lib/edamam_api_wrapper.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        recipe_list << create_recipe(hit_data)
+      end
+      return recipe_list
+    else
+      return []
+    end
+  end
+
+  def self.show_recipe(id)
+    uri_first = URI.encode("http://www.edamam.com/ontologies/edamam.owl#recipe")
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:25]: I like how you broke up the URI so you only have to pass the id portion via params.

File: test/cassettes/show_recipe.yml [hunk 4]
```diff
@@ -0,0 +1,129 @@
+---
+http_interactions:
+- request:
+    method: get
+    uri: https://api.edamam.com/search?app_id=<SLACK_TOKEN>&app_key=e5f504d20e799252972475e99ad76435&r=http://www.edamam.com/ontologies/edamam.owl%23recipe_bob
```
Comment (CheezItMan) [inline test/cassettes/show_recipe.yml]: **HEY LOOK** An App key in git!

File: test/controllers/recipes_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+require "test_helper"
+
+describe RecipesController do
+  it "should get index" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:4]: You're not testing this with search terms and invalid search terms.

File: test/lib/edamam_api_wrapper_test.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+
+describe EdamamApiWrapper do
+  it "can list recipes" do
+    VCR.use_cassette('list_recipes') do
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:5]: You should also test with a search term getting no results.

File: test/lib/edamam_api_wrapper_test.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      expect(recipe).must_respond_to :id
+      expect(recipe).must_respond_to :ingredients
+      expect(recipe).must_respond_to :dietary_info
+      expect(recipe).must_respond_to :recipe_link
+    end
+  end
+
+  it "cannot show info for a recipe with invalid id" do
+    VCR.use_cassette('show_recipe') do
+      recipe = EdamamApiWrapper.show_recipe("bob")
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:37]: `bob` a classic!

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** |
Git hygiene |  Good number of commits and good commit messages
Comprehension questions |  Check, well done
**General** |
Rails fundamentals (RESTful routing, use of named paths) | Check
Semantic HTML |  You are overusing `div` elements, consider using tags like `section`, `main`, `header`, or `article`
Errors are reported to the user |  Check, good use of `flash`
API Wrapper to handle the API requests |  Check
Controller testing |  Check, but you didn't test the search functionality well.
Lib testing |  Need better testing of search functionality
Search Functionality |  Check
List Functionality |  Check
Show individual item functionality |  Check
**Styling** |
List view shows 10 items at a time and/or has pagination |  Check
Attractive and usable UI |  It looks really nice and good on smaller screen sizes too.
**API Features** |
The App attributes Edamam |  Check
The VCR cassettes do not contain the API key |  **OOPS!**  You included your app_key in your cassettes!  **Doh!**
**External Resources** |
Link to deployed app on Heroku |  Check
**Overall** |  Nice work, you hit the major learning goals.  Areas to improve include more through testing and more semantic html.
